Shravya: Phase 4 - Tagged Logs for Young Learner Support - Frontend

ShravyaKudlu opened the pull request:
# Description:
Implements Phase 4 - Tagged Logs for Young Learner Support - Frontend
The UI will clearly identify log entries that were created or edited by a Learning Support Team member.
Features:
When an educator views a student's 'DailyLogPage', any log entry assisted by a support member will display a clear visual tag (e.g., "Assisted by: [Support Member Name]").
When a Young Learner Support member creates a log entry, the UI may or may not show a specific indicator, as they are acting on behalf of the student. The tagging is primarily for the educator's view.

Frontend Tasks:
URL: (Modify existing) '/educator/daily-log/:studentId'.
Update the 'LogItemCard.jsx' component to conditionally render an "Assisted" tag or byline based on the new API response field.
Add a new CSS module style for the "assisted" tag.
<img width="1120" height="852" alt="image" src="https://github.com/user-attachments/assets/ca5dd0e2-3cdd-45b7-9a65-ff9c2fa3e6d0" />

Figma Design provided:
<img width="1064" height="1231" alt="swappy-20251115-144455" src="https://github.com/user-attachments/assets/6b5a4a79-ce93-4e39-a548-ac763235fe46" />


## Related PRS (if any):
This frontend PR is related to the development backend.

## Main changes explained:

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to path `ActivityLogs/DailyLog`
6. verify All the buttons and functionality works as expected
7. verify this new feature works in [dark mode)
8. Reference the video below

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/d2afec5f-b3bd-4d7a-8ab4-9e5800c633d2



## Note:
Thia is is of Mock Data

ShravyaKudlu committed:
ActivityLog

ShravyaKudlu committed:
refactor: change the structure of the code

ShravyaKudlu committed:
feat: added respective assisted section

ShravyaKudlu committed:
style: Adding styles for the Components

ShravyaKudlu committed:
fix: sonarcube validation and reliability fix

beblicarl reviewed (dismissed):
This functionality works as intended and the code is great

https://www.loom.com/share/9752cb754b2c4762ab4ab55aed42f0b2

ShravyaKudlu committed:
Merge branch 'development' into Phase4/Tagged_Logs_for_Young_Learner_Support/Frontend

DeepighaJ reviewed (changes requested):
Checked out to current branch and tried to start frontend npm run start:local and getting error code: 'MODULE_NOT_FOUND'

<img width="828" height="597" alt="Image" src="https://github.com/user-attachments/assets/6d839562-0a8c-426d-a827-238917b57199" />

HemanthNidamanuru reviewed (changes requested):
Hi,
I tested this locally following the steps provided and found 2 issues:

Student name is showing "Young Learner" placeholder instead of the actual student name.
Tags need styling improvement in dark mode.

<img width="938" height="711" alt="Image" src="https://github.com/user-attachments/assets/e09c3bd2-084e-4e97-b039-efdede18b266" />

<img width="1897" height="1036" alt="Image" src="https://github.com/user-attachments/assets/da4c0fb2-534b-488a-b8d3-df035e047407" />

<img width="1901" height="1046" alt="Image" src="https://github.com/user-attachments/assets/7bded816-bbf9-4e19-9a9f-b21792835ae2" />

DeepighaJ reviewed (changes requested):
- Checked out to current branch and tested the PR locally.
- In Dark mode the Assisted by Name is not visible properly.
- Print View button is not clickable.
- Student is able to create log and submit to teacher and teacher can view it and send feedback as expected.
- Support user can log support as expected. 
- Functionality works as expected.

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/48794301-8fdf-4d6e-8c08-736eca0d772f" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/adb2435f-6afc-4de3-93c8-ecaddbd86f75" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/ec0f8c39-b99a-40f2-a3d2-2ae9c0463075" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/04916560-0bfa-4d97-9cc3-c416c996d4d4" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/5a606c99-9283-48db-881e-e10323dfe983" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/f09145a3-31be-44a9-875e-54ec28ec06ed" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/fbe0f8aa-e81e-4fab-9474-06ba516fb55c" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/b8544a70-0328-4910-8271-5305f8f36254" />

ShravyaKudlu commented:
Thanks for your feedback.

@HemanthNidamanuru  , 
1. could you please confirm again, I'm not sure what the issue is, I have pasted the requirements mentioned for more clarity, since sara is a supporter of young learners, I assumed, she would be assisting someone placeholder, so if we could integrate it to backend, maybe they would have a set have young learners they could assist, but since its not integrated, I made it like this
2. I have addressed the dark mode issue, could you please verify?
@DeepighaJ 
1. print view is not mentioned in the UI, so I removed it
2. I have addressed the dark mode issue, could you please verify

Thanks

<img width="2810" height="1223" alt="image" src="https://github.com/user-attachments/assets/858f679b-6404-49a5-a3a8-51973e35e705" />

ShravyaKudlu committed:
fix: styles and comments

ShravyaKudlu committed:
Merge branch 'development' into Phase4/Tagged_Logs_for_Young_Learner_Support/Frontend

DeepighaJ reviewed (changes requested):
Tested the PR, the dark mode issue still exist, please refer the last screenshot.

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/b20743db-a3eb-4692-82c1-36eb881a250d" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/3708a312-9637-4c86-ab64-f21231320ee2" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/95eb6c80-e03f-4083-9d14-c80c4b05dfff" />

ShravyaKudlu committed:
fix: colors

kzou55 reviewed (dismissed):
Hi Shravya, 

I ran and tested this PR locally. 

I can verify that the functionality works as expected and that there was no issue with Dark Mode. 

Testing in Light Mode

https://github.com/user-attachments/assets/3a9e4e85-0215-4daa-be03-782072210903

Testing in Dark Mode

https://github.com/user-attachments/assets/723daf85-94c1-4586-8df5-b0abcc8bfb00

DeepighaJ reviewed (dismissed):
Tested the PR locally. The dark mode issue is fixed and other functionalities working as expected. 

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/f009989d-264d-491b-8f0b-f5a368f060b1" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/b6fb7a05-cfdd-4f88-9afd-192b1b96da75" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/264f8019-e198-4eab-8415-c4af5a33421d" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/9699791a-d3ff-4a74-b780-0f99e41eedbd" />

ShravyaKudlu committed:
fix: sonarqube negate condition

ShravyaKudlu committed:
fix: sonarqube color contrast

ShravyaKudlu committed:
fix: sonarqube color contrast again
